src/util/tubemap.js

```diff
@@ -676,12 +677,57 @@ function placeReads() {
       if (!element.hasOwnProperty("y")) {
         // previous y value from pathIdx - 1 might not exist yet if that segment is also without node 
         // use previous y value from last segment with node instead 
-        let previousValidY = null;
         let lastIndex = pathIdx - 1;
-        while (previousValidY === null && lastIndex >= 0) {
-          previousValidY = reads[idx].path[lastIndex].y;
+        let previousPathWithNode;
+        while ((previousPathWithNode?.node === null || !previousPathWithNode?.node) && lastIndex >= 0) {
+          previousPathWithNode = reads[idx].path[lastIndex];
           lastIndex = lastIndex - 1;
         }
+
+        let previousValidY = previousPathWithNode?.y;
+
+        // sometimes, elements without nodes are between 2 segments going to a node we've already visited, from the same direction
+        // this means we're looping back to a node we've already been to, and we should sort in reverse
+        
+        // Find the next node in our path
+        let nextPathIndex = pathIdx + 1
+        let nextPathWithNode = reads[idx].path[nextPathIndex];
+        while ((nextPathWithNode?.node === null || !nextPathWithNode?.node) && nextPathIndex < reads[idx].path.length) {
+          nextPathWithNode = reads[idx].path[nextPathIndex];
+          nextPathIndex = nextPathIndex + 1;
+        }
+
+        let nextNodeOrder = nextPathWithNode?.order;
+        let previousNodeOrder = previousPathWithNode?.order;
+
+        // if the previous segment and the next segment is going to the same node
+        let nextNodeJustVisited = previousNodeOrder !== null && nextNodeOrder !== null && previousNodeOrder === nextNodeOrder;
+
+      
+        // a segment can also be in a loop if the next node is behind on the map
+        let nextNodeBehind = previousNodeOrder !== null && nextNodeOrder !== null && previousPathWithNode.order > nextNodeOrder;
+
+
+        reads[idx].path[pathIdx].betweenCycle = false;
```

Comment:
I don't know if the idea of "being between a cycle" means anything to me. All the visits along the path that the read takes from a node back to the same node "participate in" the cycle, but here we're specifically interested in the visits where we're in the part of the cycle that goes from right to left when laid out, I think, right? Is there another way to say this?